### ISSUE TITLE
"Add vote" action adds comment instead of voting

### ISSUE THREAD

**Issue by jostyee** [jenkinsci/gitlab-plugin#556]
### Context
- **Gitlab plugin version**: 1.4.5
- **Gitlab version**: 9.1.4
- **Jenkins version**: 2.46.2
- **Job type**: Freestyle

### Logs & Traces

```log
May 25, 2017 11:21:06 AM INFO com.dabsquared.gitlabjenkins.webhook.GitLabWebHook getDynamic

WebHook called with url: /project/gitlab_merge_request_ci

May 25, 2017 11:21:06 AM FINE com.dabsquared.gitlabjenkins.webhook.build.PushBuildAction

Push: {
  "object_kind" : "push",
  "event_name" : "push",
  "before" : "0000000000000000000000000000000000000000",
  "after" : "b41a45f767638e4515d83642ec86d75390bf935d",
  "ref" : "refs/heads/jenkins_gitlab_plugin",
  "checkout_sha" : "b41a45f767638e4515d83642ec86d75390bf935d",
  "message" : null,
  "user_id" : 14,
  "user_name" : "jostyee",
  "user_email" : "jostyee@mycompany.com",
  "user_avatar" : "https://gitlab.mycompany.com/uploads/user/avatar/14/avatar.png",
  "project_id" : 253,
  "project" : {
    "name" : "myrepo",
    "description" : "",
    "web_url" : "https://gitlab.mycompany.com/mygroup/myrepo",
    "avatar_url" : null,
    "git_ssh_url" : "git@gitlab.mycompany.com:mygroup/myrepo.git",
    "git_http_url" : "https://gitlab.mycompany.com/mygroup/myrepo.git",
    "namespace" : "mygroup",
    "visibility_level" : 10,
    "path_with_namespace" : "mygroup/myrepo",
    "default_branch" : "master",
    "homepage" : "https://gitlab.mycompany.com/mygroup/myrepo",
    "url" : "git@gitlab.mycompany.com:mygroup/myrepo.git",
    "ssh_url" : "git@gitlab.mycompany.com:mygroup/myrepo.git",
    "http_url" : "https://gitlab.mycompany.com/mygroup/myrepo.git"
  },
  "commits" : [ {
    "id" : "b41a45f767638e4515d83642ec86d75390bf935d",
    "message" : "Just for test\n",
    "timestamp" : "2017-05-25T11:19:40+08:00",
    "url" : "https://gitlab.mycompany.com/mygroup/myrepo/commit/b41a45f767638e4515d83642ec86d75390bf935d",
    "author" : {
      "name" : "jostyee",
      "email" : "jostyee@mycompany.com"
    },
    "added" : [ ],
    "modified" : [ "README.md" ],
    "removed" : [ ]
  } ],
  "total_commits_count" : 1,
  "repository" : {
    "name" : "myrepo",
    "url" : "git@gitlab.mycompany.com:mygroup/myrepo.git",
    "description" : "",
    "homepage" : "https://gitlab.mycompany.com/mygroup/myrepo",
    "git_http_url" : "https://gitlab.mycompany.com/mygroup/myrepo.git",
    "git_ssh_url" : "git@gitlab.mycompany.com:mygroup/myrepo.git",
    "visibility_level" : 10
  }
}

May 25, 2017 11:21:15 AM INFO com.dabsquared.gitlabjenkins.webhook.GitLabWebHook getDynamic

WebHook called with url: /project/gitlab_merge_request_ci

May 25, 2017 11:21:15 AM FINE com.dabsquared.gitlabjenkins.webhook.build.MergeRequestBuildAction

MergeRequest: {
  "object_kind" : "merge_request",
  "user" : {
    "name" : "jostyee",
    "username" : "jostyee",
    "avatar_url" : "https://gitlab.mycompany.com/uploads/user/avatar/14/avatar.png"
  },
  "project" : {
    "name" : "myrepo",
    "description" : "",
    "web_url" : "https://gitlab.mycompany.com/mygroup/myrepo",
    "avatar_url" : null,
    "git_ssh_url" : "git@gitlab.mycompany.com:mygroup/myrepo.git",
    "git_http_url" : "https://gitlab.mycompany.com/mygroup/myrepo.git",
    "namespace" : "mygroup",
    "visibility_level" : 10,
    "path_with_namespace" : "mygroup/myrepo",
    "default_branch" : "master",
    "homepage" : "https://gitlab.mycompany.com/mygroup/myrepo",
    "url" : "git@gitlab.mycompany.com:mygroup/myrepo.git",
    "ssh_url" : "git@gitlab.mycompany.com:mygroup/myrepo.git",
    "http_url" : "https://gitlab.mycompany.com/mygroup/myrepo.git"
  },
  "object_attributes" : {
    "id" : 1405,
    "target_branch" : "master",
    "source_branch" : "jenkins_gitlab_plugin",
    "source_project_id" : 253,
    "author_id" : 14,
    "assignee_id" : 14,
    "title" : "Just for test",
    "created_at" : "2017-05-25 11:19:56 +0800",
    "updated_at" : "2017-05-25 11:19:56 +0800",
    "milestone_id" : null,
    "state" : "opened",
    "merge_status" : "unchecked",
    "target_project_id" : 253,
    "iid" : 4,
    "description" : "",
    "position" : 0,
    "locked_at" : null,
    "updated_by_id" : null,
    "merge_error" : null,
    "merge_params" : {
      "force_remove_source_branch" : "1"
    },
    "merge_when_pipeline_succeeds" : false,
    "merge_user_id" : null,
    "merge_commit_sha" : null,
    "deleted_at" : null,
    "in_progress_merge_commit_sha" : null,
    "lock_version" : null,
    "time_estimate" : 0,
    "source" : {
      "name" : "myrepo",
      "description" : "",
      "web_url" : "https://gitlab.mycompany.com/mygroup/myrepo",
      "avatar_url" : null,
      "git_ssh_url" : "git@gitlab.mycompany.com:mygroup/myrepo.git",
      "git_http_url" : "https://gitlab.mycompany.com/mygroup/myrepo.git",
      "namespace" : "mygroup",
      "visibility_level" : 10,
      "path_with_namespace" : "mygroup/myrepo",
      "default_branch" : "master",
      "homepage" : "https://gitlab.mycompany.com/mygroup/myrepo",
      "url" : "git@gitlab.mycompany.com:mygroup/myrepo.git",
      "ssh_url" : "git@gitlab.mycompany.com:mygroup/myrepo.git",
      "http_url" : "https://gitlab.mycompany.com/mygroup/myrepo.git"
    },
    "target" : {
      "name" : "myrepo",
      "description" : "",
      "web_url" : "https://gitlab.mycompany.com/mygroup/myrepo",
      "avatar_url" : null,
      "git_ssh_url" : "git@gitlab.mycompany.com:mygroup/myrepo.git",
      "git_http_url" : "https://gitlab.mycompany.com/mygroup/myrepo.git",
      "namespace" : "mygroup",
      "visibility_level" : 10,
      "path_with_namespace" : "mygroup/myrepo",
      "default_branch" : "master",
      "homepage" : "https://gitlab.mycompany.com/mygroup/myrepo",
      "url" : "git@gitlab.mycompany.com:mygroup/myrepo.git",
      "ssh_url" : "git@gitlab.mycompany.com:mygroup/myrepo.git",
      "http_url" : "https://gitlab.mycompany.com/mygroup/myrepo.git"
    },
    "last_commit" : {
      "id" : "b41a45f767638e4515d83642ec86d75390bf935d",
      "message" : "Just for test\n",
      "timestamp" : "2017-05-25T11:19:40+08:00",
      "url" : "https://gitlab.mycompany.com/mygroup/myrepo/commit/b41a45f767638e4515d83642ec86d75390bf935d",
      "author" : {
        "name" : "jostyee",
        "email" : "jostyee@mycompany.com"
      }
    },
    "work_in_progress" : false,
    "total_time_spent" : 0,
    "human_total_time_spent" : null,
    "human_time_estimate" : null,
    "url" : "https://gitlab.mycompany.com/mygroup/myrepo/merge_requests/4",
    "action" : "open"
  },
  "labels" : [ ],
  "repository" : {
    "name" : "myrepo",
    "url" : "git@gitlab.mycompany.com:mygroup/myrepo.git",
    "description" : "",
    "homepage" : "https://gitlab.mycompany.com/mygroup/myrepo"
  },
  "assignee" : {
    "name" : "jostyee",
    "username" : "jostyee",
    "avatar_url" : "https://gitlab.mycompany.com/uploads/user/avatar/14/avatar.png"
  }
}

May 25, 2017 11:21:15 AM INFO com.dabsquared.gitlabjenkins.trigger.handler.AbstractWebHookTriggerHandler handle

gitlab_merge_request_ci triggered for merge request.

May 25, 2017 11:21:15 AM FINEST com.dabsquared.gitlabjenkins.gitlab.GitLabClientBuilder

Call GitLab:
HTTP method: POST
URL: https://gitlab.mycompany.com/api/v3/projects/253/statuses/b41a45f767638e4515d83642ec86d75390bf935d
Request headers: [
Accept = [application/json],
Content-Type = [application/x-www-form-urlencoded],
PRIVATE-TOKEN = [****FILTERED****]
]

May 25, 2017 11:21:15 AM FINEST com.dabsquared.gitlabjenkins.gitlab.GitLabClientBuilder

Got response from GitLab:
URL: https://gitlab.mycompany.com/api/v3/projects/253/statuses/b41a45f767638e4515d83642ec86d75390bf935d
Status: 201 Created
Response headers: [
Cache-Control = [max-age=0, private, must-revalidate],
Connection = [keep-alive],
Content-Length = [549],
Content-Type = [application/json],
Date = [Thu, 25 May 2017 03:19:57 GMT],
Etag = [W/"3f13b90df154dd5eb1f891e638de219b"],
Strict-Transport-Security = [max-age=31536000;],
Vary = [Origin],
X-Request-Id = [9244e3db-2d06-4bb2-a484-0f908f7b4209],
X-Runtime = [0.082755]
]
Response body: {
  "id" : 1024,
  "sha" : "b41a45f767638e4515d83642ec86d75390bf935d",
  "ref" : "jenkins_gitlab_plugin",
  "status" : "pending",
  "name" : "jenkins",
  "target_url" : "https://ci.mycompany.com/job/gitlab_merge_request_ci/170/",
  "description" : null,
  "created_at" : "2017-05-25T11:19:57.363+08:00",
  "started_at" : null,
  "finished_at" : null,
  "allow_failure" : false,
  "coverage" : null,
  "author" : {
    "name" : "marvin_bot",
    "username" : "marvin",
    "id" : 42,
    "state" : "active",
    "avatar_url" : "https://gitlab.mycompany.com/uploads/user/avatar/42/avatar.png",
    "web_url" : "https://gitlab.mycompany.com/marvin"
  }
}

May 25, 2017 11:21:25 AM INFO com.dabsquared.gitlabjenkins.util.CommitStatusUpdater retrieveGitlabProjectIds

Retrieving gitlab project ids

May 25, 2017 11:21:25 AM FINEST com.dabsquared.gitlabjenkins.gitlab.GitLabClientBuilder

Call GitLab:
HTTP method: GET
URL: https://gitlab.mycompany.com/api/v3/projects/253/repository/commits/b41a45f767638e4515d83642ec86d75390bf935d
Request headers: [
Accept = [application/json],
PRIVATE-TOKEN = [****FILTERED****]
]

May 25, 2017 11:21:26 AM FINEST com.dabsquared.gitlabjenkins.gitlab.GitLabClientBuilder

Got response from GitLab:
URL: https://gitlab.mycompany.com/api/v3/projects/253/repository/commits/b41a45f767638e4515d83642ec86d75390bf935d
Status: 200 OK
Response headers: [
Cache-Control = [max-age=0, private, must-revalidate],
Connection = [keep-alive],
Content-Length = [512],
Content-Type = [application/json],
Date = [Thu, 25 May 2017 03:20:07 GMT],
Etag = [W/"e9e34608e936fc16f25e5b9e149ede65"],
Strict-Transport-Security = [max-age=31536000;],
Vary = [Origin],
X-Request-Id = [a43d039f-0035-4f98-bbf8-f6422594ba1f],
X-Runtime = [0.117879]
]
Response body: {
  "id" : "b41a45f767638e4515d83642ec86d75390bf935d",
  "short_id" : "b41a45f7",
  "title" : "Just for test",
  "created_at" : "2017-05-25T11:19:40.000+08:00",
  "parent_ids" : [ "f335e86fba3533d6d7e56fa83ff5037d4c1b1e76" ],
  "message" : "Just for test\n",
  "author_name" : "jostyee",
  "author_email" : "jostyee@mycompany.com",
  "authored_date" : "2017-05-25T11:19:40.000+08:00",
  "committer_name" : "jostyee",
  "committer_email" : "jostyee@mycompany.com",
  "committed_date" : "2017-05-25T11:19:40.000+08:00",
  "stats" : {
    "additions" : 2,
    "deletions" : 0,
    "total" : 2
  },
  "status" : "pending"
}

May 25, 2017 11:21:26 AM FINEST com.dabsquared.gitlabjenkins.gitlab.GitLabClientBuilder

Call GitLab:
HTTP method: POST
URL: https://gitlab.mycompany.com/api/v3/projects/253/statuses/b41a45f767638e4515d83642ec86d75390bf935d
Request headers: [
Accept = [application/json],
Content-Type = [application/x-www-form-urlencoded],
PRIVATE-TOKEN = [****FILTERED****]
]

May 25, 2017 11:21:26 AM FINEST com.dabsquared.gitlabjenkins.gitlab.GitLabClientBuilder

Got response from GitLab:
URL: https://gitlab.mycompany.com/api/v3/projects/253/statuses/b41a45f767638e4515d83642ec86d75390bf935d
Status: 201 Created
Response headers: [
Cache-Control = [max-age=0, private, must-revalidate],
Connection = [keep-alive],
Content-Length = [576],
Content-Type = [application/json],
Date = [Thu, 25 May 2017 03:20:07 GMT],
Etag = [W/"a8ace1b44c6e0b3578472f615e8d1980"],
Strict-Transport-Security = [max-age=31536000;],
Vary = [Origin],
X-Request-Id = [ec4f274d-c5ae-4ad3-be17-94f704ed3b1e],
X-Runtime = [0.070860]
]
Response body: {
  "id" : 1024,
  "sha" : "b41a45f767638e4515d83642ec86d75390bf935d",
  "ref" : "jenkins_gitlab_plugin",
  "status" : "running",
  "name" : "jenkins",
  "target_url" : "https://ci.mycompany.com/job/gitlab_merge_request_ci/170/",
  "description" : null,
  "created_at" : "2017-05-25T11:19:57.363+08:00",
  "started_at" : "2017-05-25T11:20:07.886+08:00",
  "finished_at" : null,
  "allow_failure" : false,
  "coverage" : null,
  "author" : {
    "name" : "marvin_bot",
    "username" : "marvin",
    "id" : 42,
    "state" : "active",
    "avatar_url" : "https://gitlab.mycompany.com/uploads/user/avatar/42/avatar.png",
    "web_url" : "https://gitlab.mycompany.com/marvin"
  }
}

May 25, 2017 11:21:49 AM FINEST com.dabsquared.gitlabjenkins.gitlab.GitLabClientBuilder

Call GitLab:
HTTP method: POST
URL: https://gitlab.mycompany.com/api/v3/projects/253/merge_requests/1405/notes
Request headers: [
Accept = [application/json],
Content-Type = [application/x-www-form-urlencoded],
PRIVATE-TOKEN = [****FILTERED****]
]

May 25, 2017 11:21:49 AM FINEST com.dabsquared.gitlabjenkins.gitlab.GitLabClientBuilder

Got response from GitLab:
URL: https://gitlab.mycompany.com/api/v3/projects/253/merge_requests/1405/notes
Status: 201 Created
Response headers: [
Cache-Control = [max-age=0, private, must-revalidate],
Connection = [keep-alive],
Content-Length = [605],
Content-Type = [application/json],
Date = [Thu, 25 May 2017 03:20:30 GMT],
Etag = [W/"aa0afe265ebc68a22561f143ad7dc02b"],
Strict-Transport-Security = [max-age=31536000;],
Vary = [Origin],
X-Request-Id = [5cac3e5c-5296-49b2-9c37-d7ef4055f5fa],
X-Runtime = [0.128564]
]
Response body: {
  "id" : 11795,
  "body" : ":white_check_mark: Jenkins Build SUCCESS\n\nResults available at: [Jenkins [gitlab_merge_request_ci #170]](https://ci.mycompany.com/job/gitlab_merge_request_ci/170/)",
  "attachment" : null,
  "author" : {
    "name" : "marvin_bot",
    "username" : "marvin",
    "id" : 42,
    "state" : "active",
    "avatar_url" : "https://gitlab.mycompany.com/uploads/user/avatar/42/avatar.png",
    "web_url" : "https://gitlab.mycompany.com/marvin"
  },
  "created_at" : "2017-05-25T11:20:30.816+08:00",
  "updated_at" : "2017-05-25T11:20:30.816+08:00",
  "system" : false,
  "noteable_id" : 1405,
  "noteable_type" : "MergeRequest",
  "upvote?" : false,
  "downvote?" : false
}

May 25, 2017 11:21:49 AM FINEST com.dabsquared.gitlabjenkins.gitlab.GitLabClientBuilder

Call GitLab:
HTTP method: POST
URL: https://gitlab.mycompany.com/api/v3/projects/253/merge_requests/1405/notes
Request headers: [
Accept = [application/json],
Content-Type = [application/x-www-form-urlencoded],
PRIVATE-TOKEN = [****FILTERED****]
]

May 25, 2017 11:21:49 AM FINEST com.dabsquared.gitlabjenkins.gitlab.GitLabClientBuilder

Got response from GitLab:
URL: https://gitlab.mycompany.com/api/v3/projects/253/merge_requests/1405/notes
Status: 201 Created
Response headers: [
Cache-Control = [max-age=0, private, must-revalidate],
Connection = [keep-alive],
Content-Length = [441],
Content-Type = [application/json],
Date = [Thu, 25 May 2017 03:20:30 GMT],
Etag = [W/"752577d69096c10c3037f66e7844aad8"],
Strict-Transport-Security = [max-age=31536000;],
Vary = [Origin],
X-Request-Id = [30ed707c-7518-4e07-aa6e-918cdd76ec2e],
X-Runtime = [0.106786]
]
Response body: {
  "id" : 11796,
  "body" : ":+1:",
  "attachment" : null,
  "author" : {
    "name" : "marvin_bot",
    "username" : "marvin",
    "id" : 42,
    "state" : "active",
    "avatar_url" : "https://gitlab.mycompany.com/uploads/user/avatar/42/avatar.png",
    "web_url" : "https://gitlab.mycompany.com/marvin"
  },
  "created_at" : "2017-05-25T11:20:30.935+08:00",
  "updated_at" : "2017-05-25T11:20:30.935+08:00",
  "system" : false,
  "noteable_id" : 1405,
  "noteable_type" : "MergeRequest",
  "upvote?" : false,
  "downvote?" : false
}

May 25, 2017 11:21:49 AM INFO com.dabsquared.gitlabjenkins.util.CommitStatusUpdater retrieveGitlabProjectIds

Retrieving gitlab project ids

May 25, 2017 11:21:49 AM FINEST com.dabsquared.gitlabjenkins.gitlab.GitLabClientBuilder

Call GitLab:
HTTP method: GET
URL: https://gitlab.mycompany.com/api/v3/projects/253/repository/commits/b41a45f767638e4515d83642ec86d75390bf935d
Request headers: [
Accept = [application/json],
PRIVATE-TOKEN = [****FILTERED****]
]

May 25, 2017 11:21:49 AM FINEST com.dabsquared.gitlabjenkins.gitlab.GitLabClientBuilder

Got response from GitLab:
URL: https://gitlab.mycompany.com/api/v3/projects/253/repository/commits/b41a45f767638e4515d83642ec86d75390bf935d
Status: 200 OK
Response headers: [
Cache-Control = [max-age=0, private, must-revalidate],
Connection = [keep-alive],
Content-Length = [512],
Content-Type = [application/json],
Date = [Thu, 25 May 2017 03:20:31 GMT],
Etag = [W/"ae0b7ad2b3408d7b20b61c09b1503fb0"],
Strict-Transport-Security = [max-age=31536000;],
Vary = [Origin],
X-Request-Id = [8a5df4f4-1913-4b5c-bf4a-c920cc14b174],
X-Runtime = [0.094259]
]
Response body: {
  "id" : "b41a45f767638e4515d83642ec86d75390bf935d",
  "short_id" : "b41a45f7",
  "title" : "Just for test",
  "created_at" : "2017-05-25T11:19:40.000+08:00",
  "parent_ids" : [ "f335e86fba3533d6d7e56fa83ff5037d4c1b1e76" ],
  "message" : "Just for test\n",
  "author_name" : "jostyee",
  "author_email" : "jostyee@mycompany.com",
  "authored_date" : "2017-05-25T11:19:40.000+08:00",
  "committer_name" : "jostyee",
  "committer_email" : "jostyee@mycompany.com",
  "committed_date" : "2017-05-25T11:19:40.000+08:00",
  "stats" : {
    "additions" : 2,
    "deletions" : 0,
    "total" : 2
  },
  "status" : "running"
}

May 25, 2017 11:21:49 AM FINEST com.dabsquared.gitlabjenkins.gitlab.GitLabClientBuilder

Call GitLab:
HTTP method: POST
URL: https://gitlab.mycompany.com/api/v3/projects/253/statuses/b41a45f767638e4515d83642ec86d75390bf935d
Request headers: [
Accept = [application/json],
Content-Type = [application/x-www-form-urlencoded],
PRIVATE-TOKEN = [****FILTERED****]
]

May 25, 2017 11:21:49 AM FINEST com.dabsquared.gitlabjenkins.gitlab.GitLabClientBuilder

Got response from GitLab:
URL: https://gitlab.mycompany.com/api/v3/projects/253/statuses/b41a45f767638e4515d83642ec86d75390bf935d
Status: 201 Created
Response headers: [
Cache-Control = [max-age=0, private, must-revalidate],
Connection = [keep-alive],
Content-Length = [603],
Content-Type = [application/json],
Date = [Thu, 25 May 2017 03:20:31 GMT],
Etag = [W/"5e2face9708ba219742104b35f9dca80"],
Strict-Transport-Security = [max-age=31536000;],
Vary = [Origin],
X-Request-Id = [3f1226ea-a5fb-4477-ae76-12274cbdb83b],
X-Runtime = [0.078891]
]
Response body: {
  "id" : 1024,
  "sha" : "b41a45f767638e4515d83642ec86d75390bf935d",
  "ref" : "jenkins_gitlab_plugin",
  "status" : "success",
  "name" : "jenkins",
  "target_url" : "https://ci.mycompany.com/job/gitlab_merge_request_ci/170/",
  "description" : null,
  "created_at" : "2017-05-25T11:19:57.363+08:00",
  "started_at" : "2017-05-25T11:20:07.886+08:00",
  "finished_at" : "2017-05-25T11:20:31.148+08:00",
  "allow_failure" : false,
  "coverage" : null,
  "author" : {
    "name" : "marvin_bot",
    "username" : "marvin",
    "id" : 42,
    "state" : "active",
    "avatar_url" : "https://gitlab.mycompany.com/uploads/user/avatar/42/avatar.png",
    "web_url" : "https://gitlab.mycompany.com/marvin"
  }
}
```

### Problem description

_Describe your problem in a meaningful way_:
- what were you doing (simple push, merge request, MR with fork, ...)

    Trigger a Jenkins job with new GitLab MR, which include a post-build action "Add vote for build status on GitLab merge requests"

- what was expected

    Add count to the thumbs up/down icon on the MR

- what occurred finally

    The user add a comment with thumbs up/down under the MR
    <img width="130" alt="screen shot 2017-05-19 at 8 11 29 am" src="https://cloud.githubusercontent.com/assets/88564/26228498/e78b5f0a-3c6c-11e7-894a-ccd71fa3570d.png">
    <img width="138" alt="screen shot 2017-05-19 at 8 11 38 am" src="https://cloud.githubusercontent.com/assets/88564/26228499/e78b98ee-3c6c-11e7-9af9-e0f5c39b772c.png">

**Comment by omehegan** [comment]
I have a feeling that maybe the voting API has changed in GitLab 9.x. Was this working for you in a previous version of GitLab?

**Comment by jostyee** [comment]
@omehegan It worked when it's 8.x, I'm sorry I can't remember which specific version it changed.

**Comment by omehegan** [comment]
I'm guessing something changed in GitLab 9.0.

**Comment by deepy** [comment]
I've started to work on this, the change that caused this was introduced in 8.9

You now have to use [award_emoji](https://docs.gitlab.com/ee/api/award_emoji.html#award-a-new-emoji) instead.

Although I'm unsure of how to use [delete_emoji](https://docs.gitlab.com/ee/api/award_emoji.html#delete-an-award-emoji) since it requires the award_id. Which can be retrieved by a GET to [award_emoji](https://docs.gitlab.com/ee/api/award_emoji.html#list-an-awardable-39-s-award-emoji)

Ok I found the getCurrentUser() but this just feels silly and kinda roundabout. Check who I am, then check the merge request to see if I've already added the opposite award, then add an award.

**Comment by omehegan** [comment]
Oh yeah, award emoji. I had to deal with this when I was writing some code to check for votes/thumbs up on MRs. Thanks for working on this. I would say don't worry about backwards compatibility to GitLab 8.x, at this point it is several years old and IMO we should only maintain support for the current and previous major version, so 9.x and 10.x currently.

**Comment by deepy** [comment]
Luckily the v3 and v4 api both have this, so I'm just going to make sure both handle it

**Comment by deepy** [comment]
Alright, so the pull request is now ready to be merged.

**Comment by alejandrovelez7** [comment]
I'm getting similar behavior using this in a pipeline script. 

    addGitLabMRComment comment: result == 'success' ? ':+1:' : ':-1:'

How can we accomplish thumbs down on the merge request?

**Comment by deepy** [comment]
If I'm reading `src/main/java/com/dabsquared/gitlabjenkins/workflow/` right then it seems like we'll need an additional step for the 'add emoji/vote'. So maybe this should go into a separate issue.

In your specific case however, keep in mind that if your script runs twice on the same GitLabMR it could potentially add both a `thumbsup` and a `thumbsdown`.

**Comment by alejandrovelez7** [comment]
That's a good point. 

Also I thought it would be similar to Github's approval, I guess it doesn't really matter for Gitlab CE as long as the team agrees on the 'approval' comment. Thanks!

**Comment by deepy** [comment]
I guess if #748 gets merged we could extract that logic and use it for a `vote` step.

**Comment by omehegan** [comment]
Since I haven't been able to get anyone else to review the PR, I built a snapshot with this change: http://repo.jenkins-ci.org/snapshots/org/jenkins-ci/plugins/gitlab-plugin/1.5.6-SNAPSHOT/gitlab-plugin-1.5.6-20180628.021351-4.hpi  Can you guys test this on your systems and let me know if you find any issues? If not I will merge.

**Comment by omehegan** [comment]
This was released in version 1.5.9 of the plugin.
